packages/itwin/geo-tools/src/components/GeoAddressSearch.tsx

```diff
@@ -3,111 +3,80 @@
 *--------------------------------------------------------------------------------------------*/
 
-import "./GeoAddressSearch.scss";
+import * as React from "react";
+import { SvgCloseSmall, SvgSearch } from "@itwin/itwinui-icons-react";
+import type { SelectOption } from "@itwin/itwinui-react";
+import { ComboBox, IconButton } from "@itwin/itwinui-react";
 
-import type { AutoSuggestData, CommonProps} from "@itwin/core-react";
-import { AutoSuggest, WebFontIcon } from "@itwin/core-react";
-import type { AddressProvider} from "../AddressProvider";
+import type { AddressData, AddressProvider } from "../AddressProvider";
 import { BingAddressProvider } from "../AddressProvider";
-import * as React from "react";
-import type * as ReactAutosuggest from "react-autosuggest";
 import { IModelGeoView } from "../IModelGeoView";
 import { GeoTools } from "../GeoTools";
+import "./GeoAddressSearch.scss";
 
-export interface GeoAddressSearchProps extends CommonProps {
+export interface GeoAddressSearchProps {
   /** Address provider object */
   provider?: AddressProvider;
   /** Indicates whether to set focus to the input element (default to true)*/
   setFocus?: boolean;
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
 export function GeoAddressSearch(props: GeoAddressSearchProps) {
   const [inputValue, setInputValue] = React.useState("");
+  const [options, setOptions] = React.useState<SelectOption<string>[]>([]);
 
   // `React.useMemo' is used avoid creating new object on each render cycle
-  const addressProvider = React.useMemo(() => (props.provider ?? new BingAddressProvider()), [props.provider]);
+  const addressProvider = React.useMemo(() => props.provider ?? new BingAddressProvider(), [props.provider]);
 
-  const onSuggestionSelected = (selected: AutoSuggestData) => {
-    setInputValue(selected.label);
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    IModelGeoView.locateAddress(selected.label);
+  const onAddressSelected = (selected: string) => {
+    setInputValue(selected);
+    IModelGeoView.locateAddress(selected);
   };
 
-  const getAutoSuggestDataFunc = async (value: string) => {
-
-    const data: AutoSuggestData[] = [];
+  const getAddressesFunc = async (value: string): Promise<AddressData[]> => {
     const viewBBox = IModelGeoView.getFrustumLonLatBBox();
-    if (viewBBox) {
-      const addresses = await addressProvider.getAddresses(value, viewBBox);
-
-      addresses.forEach((addr) => { data.push({ value: addr.addressLine, label: addr.formattedAddress }); });
+    if (viewBBox && value) {
+      return await addressProvider.getAddresses(value, viewBBox);
     }
-    return Promise.resolve(data);
+    return [];
   };
 
+  void React.useEffect(() => {
+    void (async () => {
+      const items = await getAddressesFunc(inputValue);
+      setOptions(items.map((value) => ({ label: value.formattedAddress ?? "", value: value.formattedAddress ?? "" })));
+    })();
```

Comment:
should this be wrapped in a try/catch?

Reply:
No I don't think so - I already handled the issue where the returned address is undefined